pkg/virt-controller/watch/ovm.go

```diff
@@ -0,0 +1,712 @@
+			return nil, err
+		}
+		if fresh.ObjectMeta.UID != OVM.ObjectMeta.UID {
+			return nil, fmt.Errorf("original OfflineVirtualMachine %v/%v is gone: got uid %v, wanted %v", OVM.Namespace, OVM.Name, fresh.UID, OVM.UID)
+		}
+		return fresh, nil
+	})
+	cm := controller.NewVirtualMachineControllerRefManager(controller.RealVirtualMachineControl{Clientset: c.clientset}, OVM, nil, virtv1.OfflineVirtualMachineGroupVersionKind, canAdoptFunc)
+
+	var vm *virtv1.VirtualMachine
+	vmObj, exist, err := c.vmInformer.GetStore().GetByKey(ovmKey)
+	if err != nil {
+		logger.Reason(err).Error("Failed to fetch vm for namespace from cache.")
+		return err
+	}
+	if !exist {
+		logger.Infof("VM not found in cache %s", key)
+		vm = nil
+	} else {
+		vm = vmObj.(*virtv1.VirtualMachine)
+
+		vm, err = cm.ClaimVirtualMachineByName(vm)
+		if err != nil {
+			return err
+		}
+	}
+
+	var createErr error
+
+	// Scale up or down, if all expected creates and deletes were report by the listener
+	if needsSync && OVM.ObjectMeta.DeletionTimestamp == nil {
+		logger.Infof("Creating or the VM: %t", OVM.Spec.Running)
+		createErr = c.startStop(OVM, vm)
+	}
+
+	// If the controller is going to be deleted and the orphan finalizer is the next one, release the VMs. Don't update the status
+	// TODO: Workaround for https://github.com/kubernetes/kubernetes/issues/56348, remove it once it is fixed
+	if OVM.ObjectMeta.DeletionTimestamp != nil && controller.HasFinalizer(OVM, v1.FinalizerOrphanDependents) {
+		return c.orphan(cm, vm)
+	}
+
+	if createErr != nil {
+		logger.Reason(err).Error("Scaling the OfflineVirtualMachine failed.")
+	}
+
+	err = c.updateStatus(OVM.DeepCopy(), vm, createErr)
+	if err != nil {
+		logger.Reason(err).Error("Updating the OfflineVirtualMachine status failed.")
+	}
+
+	return err
+}
+
+// orphan removes the owner reference of all VMs which are owned by the controller instance.
+// Workaround for https://github.com/kubernetes/kubernetes/issues/56348 to make no-cascading deletes possible
+// We don't have to remove the finalizer. This part of the gc is not affected by the mentioned bug
+// TODO +pkotas unify with replicasets. This function can be the same
+func (c *OVMController) orphan(cm *controller.VirtualMachineControllerRefManager, vm *virtv1.VirtualMachine) error {
+	if vm == nil {
+		return nil
+	}
+
+	errChan := make(chan error, 1)
+
+	go func(vm *virtv1.VirtualMachine) {
+		err := cm.ReleaseVirtualMachine(vm)
+		if err != nil {
+			errChan <- err
+		}
+	}(vm)
+
+	select {
+	case err := <-errChan:
+		return err
+	default:
+	}
+	return nil
+}
+
+func (c *OVMController) startStop(ovm *virtv1.OfflineVirtualMachine, vm *virtv1.VirtualMachine) error {
+
+	ovmKey, err := controller.KeyFunc(ovm)
+	if err != nil {
+		log.Log.Object(ovm).Reason(err).Error("Failed to extract ovmKey from OfflineVirtualMachine.")
+		return nil
+	}
+
+	log.Log.Object(ovm).Infof("Creating the vm since: %t", ovm.Spec.Running)
+
+	if ovm.Spec.Running == true {
+		if vm != nil {
```

Comment:
You have to check here if the vm is in a final state (failed, succeeded, **not** unknown). Since the VM should run, you wil have to delete it and wait for the delete confirmation. Otherwise you can never start a VM again, once it failed or the vm was shut down from inside.